Unittest test_pretty.py::test_pretty fails on i586 systems (openSUSE Tumbleweed)

With the latest version bump I noticed that one test fails on i586 systems.
At least versions 6.0.0 and 5.2.1 are affected. The same test runs fine on x86_64 systems.

Unittest `test_pretty.py::test_pretty` fails:
```
[   10s] =================================== FAILURES ===================================
[   10s] _________________________________ test_pretty __________________________________
[   10s] 
[   10s]     def test_pretty():
[   10s]         test = {
[   10s]             "foo": [1, 2, 3, {4, 5, 6, (7, 8, 9)}, {}],
[   10s]             "bar": {"egg": "baz", "words": ["Hello World"] * 10},
[   10s]             False: "foo",
[   10s]             True: "",
[   10s]             "text": ("Hello World", "foo bar baz egg"),
[   10s]         }
[   10s]     
[   10s]         result = pretty_repr(test, max_width=80)
[   10s]         print(result)
[   10s]         print(repr(result))
[   10s]         expected = "{\n    'foo': [1, 2, 3, {(7, 8, 9), 4, 5, 6}, {}],\n    'bar': {\n        'egg': 'baz',\n        'words': [\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World'\n        ]\n    },\n    False: 'foo',\n    True: '',\n    'text': ('Hello World', 'foo bar baz egg')\n}"
[   10s] >       assert result == expected
[   10s] E       assert ('{\n'\n "    'foo': [1, 2, 3, {4, 5, 6, (7, 8, 9)}, {}],\n"\n "    'bar': {\n"\n "        'egg': 'baz',\n"\n "        'words': [\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World'\n"\n '        ]\n'\n '    },\n'\n "    False: 'foo',\n"\n "    True: '',\n"\n "    'text': ('Hello World', 'foo bar baz egg')\n"\n '}') == ('{\n'\n "    'foo': [1, 2, 3, {(7, 8, 9), 4, 5, 6}, {}],\n"\n "    'bar': {\n"\n "        'egg': 'baz',\n"\n "        'words': [\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World',\n"\n "            'Hello World'\n"\n '        ]\n'\n '    },\n'\n "    False: 'foo',\n"\n "    True: '',\n"\n "    'text': ('Hello World', 'foo bar baz egg')\n"\n '}')
[   10s] E           {
[   10s] E         -     'foo': [1, 2, 3, {(7, 8, 9), 4, 5, 6}, {}],
[   10s] E         ?                                ---------
[   10s] E         +     'foo': [1, 2, 3, {4, 5, 6, (7, 8, 9)}, {}],
[   10s] E         ?                       +++++++++
[   10s] E               'bar': {
[   10s] E                   'egg': 'baz',
[   10s] E                   'words': [
[   10s] E                       'Hello World',
[   10s] E                       'Hello World',
[   10s] E                       'Hello World',
[   10s] E                       'Hello World',
[   10s] E                       'Hello World',
[   10s] E                       'Hello World',
[   10s] E                       'Hello World',
[   10s] E                       'Hello World',
[   10s] E                       'Hello World',
[   10s] E                       'Hello World'
[   10s] E                   ]
[   10s] E               },
[   10s] E               False: 'foo',
[   10s] E               True: '',
[   10s] E               'text': ('Hello World', 'foo bar baz egg')
[   10s] E           }
[   10s] 
[   10s] tests/test_pretty.py:32: AssertionError
[   10s] ----------------------------- Captured stdout call -----------------------------
[   10s] {
[   10s]     'foo': [1, 2, 3, {4, 5, 6, (7, 8, 9)}, {}],
[   10s]     'bar': {
[   10s]         'egg': 'baz',
[   10s]         'words': [
[   10s]             'Hello World',
[   10s]             'Hello World',
[   10s]             'Hello World',
[   10s]             'Hello World',
[   10s]             'Hello World',
[   10s]             'Hello World',
[   10s]             'Hello World',
[   10s]             'Hello World',
[   10s]             'Hello World',
[   10s]             'Hello World'
[   10s]         ]
[   10s]     },
[   10s]     False: 'foo',
[   10s]     True: '',
[   10s]     'text': ('Hello World', 'foo bar baz egg')
[   10s] }
[   10s] "{\n    'foo': [1, 2, 3, {4, 5, 6, (7, 8, 9)}, {}],\n    'bar': {\n        'egg': 'baz',\n        'words': [\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World'\n        ]\n    },\n    False: 'foo',\n    True: '',\n    'text': ('Hello World', 'foo bar baz egg')\n}"
[   10s] =========================== short test summary info ============================
[   10s] FAILED tests/test_pretty.py::test_pretty - assert ('{\n'\n "    'foo': [1, 2,...
[   10s] ======================== 1 failed, 291 passed in 2.83s =========================

```


**i586 full buildlog (with failures):**
https://build.opensuse.org/package/live_build_log/devel:languages:python/python-rich/openSUSE_Tumbleweed/i586
local copy
[openSUSE_Tumbleweed-i586-python-rich-buildlog.txt](https://github.com/willmcgugan/rich/files/5125817/openSUSE_Tumbleweed-i586-python-rich-buildlog.txt)
**x86_64 full buildlog**
https://build.opensuse.org/package/live_build_log/devel:languages:python/python-rich/openSUSE_Tumbleweed/x86_64

--- a/tests/test_pretty.py
+++ b/tests/test_pretty.py
@@ -18,7 +18,7 @@ def test_install():
 
 def test_pretty():
     test = {
-        "foo": [1, 2, 3, {4, 5, 6, (7, 8, 9)}, {}],
+        "foo": [1, 2, 3, (4, 5, {6}, 7, 8, {9}), {}],
         "bar": {"egg": "baz", "words": ["Hello World"] * 10},
         False: "foo",
         True: "",
@@ -28,7 +28,7 @@ def test_pretty():
     result = pretty_repr(test, max_width=80)
     print(result)
     print(repr(result))
-    expected = "{\n    'foo': [1, 2, 3, {(7, 8, 9), 4, 5, 6}, {}],\n    'bar': {\n        'egg': 'baz',\n        'words': [\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World'\n        ]\n    },\n    False: 'foo',\n    True: '',\n    'text': ('Hello World', 'foo bar baz egg')\n}"
+    expected = "{\n    'foo': [1, 2, 3, (4, 5, {6}, 7, 8, {9}), {}],\n    'bar': {\n        'egg': 'baz',\n        'words': [\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World',\n            'Hello World'\n        ]\n    },\n    False: 'foo',\n    True: '',\n    'text': ('Hello World', 'foo bar baz egg')\n}"
     assert result == expected
 
 

